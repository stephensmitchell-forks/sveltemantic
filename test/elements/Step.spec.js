import { render, cleanup, fireEvent, waitForElement } from '@testing-library/svelte'
import Step from '../../src/elements/Step.svelte'

beforeEach(() => {
    testId = id('step')
    cleanup()
})

describe('Step.svelte', () => {
    test('Step with default props', () => {
        const { getByTestId } = render(Step, { props: propsWithTestId() })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step')
        expect(getByTestId(testId)).toBeInstanceOf(HTMLDivElement)
    })

    test('Step with link prop', () => {
        const href = 'https://github.com/titans-inc/sveltemantic'
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                link: href
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveAttribute('href', href)
        expect(getByTestId(testId)).toBeInstanceOf(HTMLAnchorElement)
    })

    test('Step with active prop', () => {
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                active: true
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step', 'active')
    })

    test('Step with completed prop', () => {
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                completed: true
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step', 'completed')
    })

    test('Step with disabled prop', () => {
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                disabled: true
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step', 'disabled')
    })

    test('Step with additional classes and attributes', () => {
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                class: 'clearfix command'
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step', 'clearfix', 'command')
        expect(getByTestId(testId)).toHaveAttribute('data-testid', testId)
    })

    test('Step with custom actions', () => {
        const { getByTestId } = render(Step, {
            props: propsWithTestId({
                use: [
                    node => node.classList.add('active'),
                    node => node.setAttribute('data-step', '#steps')
                ]
            })
        })

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(getByTestId(testId)).toHaveClass('step', 'active')
        expect(getByTestId(testId)).toHaveAttribute('data-step', '#steps')
    })

    test('Step with dom events', async () => {
        const { component, getByTestId } = render(Step, {
            props: propsWithTestId({
                link: 'https://google.com'
            })
        })

        component.$on('mousedown', event => {
            event.target.setAttribute('mousedown', true)
        })

        fireEvent.mouseDown(getByTestId(testId))

        const div = await waitForElement(() => getByTestId(testId))

        expect(ignoreTestId(getByTestId(testId))).toMatchSnapshot()
        expect(div).toHaveAttribute('mousedown', 'true')
    })

    test('Step with multiple props and events snapshot', async () => {
        const { component, getByTestId } = render(Step, {
            props: propsWithTestId({
                class: 'extra',
                completed: true,
                disabled: true,
                link: 'https://github.com/titans-inc/sveltemantic',
                use: [
                    node => node.classList.add('orange'),
                    node => node.setAttribute('data-tag', 'card')
                ]
            })
        })

        component.$on('cut', event => {
            event.target.setAttribute('cut', true)
        })

        component.$on('copy', event => {
            event.target.setAttribute('copy', true)
        })

        fireEvent.cut(getByTestId(testId))
        fireEvent.copy(getByTestId(testId))

        const div = await waitForElement(() => getByTestId(testId))

        expect(ignoreTestId(div)).toMatchSnapshot()
    })
})
