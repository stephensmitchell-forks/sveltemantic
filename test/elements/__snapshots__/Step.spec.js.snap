// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step.svelte Step with active prop 1`] = `"<div class=\\"step active\\" data-testid=\\"any\\"></div>"`;

exports[`Step.svelte Step with additional classes and attributes 1`] = `"<div class=\\"step clearfix command\\" data-testid=\\"any\\"></div>"`;

exports[`Step.svelte Step with completed prop 1`] = `"<div class=\\"step completed\\" data-testid=\\"any\\"></div>"`;

exports[`Step.svelte Step with custom actions 1`] = `"<div class=\\"step active\\" data-testid=\\"any\\" data-step=\\"#steps\\"></div>"`;

exports[`Step.svelte Step with default props 1`] = `"<div class=\\"step\\" data-testid=\\"any\\"></div>"`;

exports[`Step.svelte Step with disabled prop 1`] = `"<div class=\\"step disabled\\" data-testid=\\"any\\"></div>"`;

exports[`Step.svelte Step with dom events 1`] = `"<a href=\\"https://google.com\\" class=\\"step\\" data-testid=\\"any\\" mousedown=\\"true\\"></a>"`;

exports[`Step.svelte Step with link prop 1`] = `"<a href=\\"https://github.com/titans-inc/sveltemantic\\" class=\\"step\\" data-testid=\\"any\\"></a>"`;

exports[`Step.svelte Step with multiple props and events snapshot 1`] = `"<a href=\\"https://github.com/titans-inc/sveltemantic\\" class=\\"step extra completed disabled orange\\" data-testid=\\"any\\" data-tag=\\"card\\" cut=\\"true\\" copy=\\"true\\"></a>"`;
